src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -242,7 +276,6 @@ private static void EmitFindFirstChar(IndentedTextWriter writer, RegexMethod rm,
         {
             RegexOptions options = (RegexOptions)rm.Options;
             RegexCode code = rm.Code;
-            bool rtl = code.RightToLeft;
```

Comment:
I'm sure I'm missing something obvious, but why did we have all of this code split specific to rtl code emittion if RegexNode did not support RTL anyway? Was this just dead code?

Reply:
The engine supported RightToLeft; the Go implementation would just fall back to using the older code generation for it.  This PR deletes that older code generation, so we are then only able to interpret RightToLeft.